### PR TITLE
Move embedder jar file processing to GN.

### PR DESCRIPTION
**Pull request by godofredoc**
Recipes currently contain a lot of logic to manipulate jar file names
and creating the expected folder structures. This PR moves the logic to
process embedder android artifacts to GN.

Bug: https://github.com/flutter/flutter/issues/81855

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: build/android_artifacts.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+"""Copies and renames android artifacts."""
+
+
```
Comment (blasten) [inline build/android_artifacts.py:9]:
```suggestion
```

File: build/android_artifacts.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import sys
+import shutil
+import os
+import argparse
+
```
Comment (blasten) [inline build/android_artifacts.py:15]:
```suggestion
```

File: build/android_artifacts.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# found in the LICENSE file.
+
+"""Copies and renames android artifacts."""
+
+
+
+import sys
+import shutil
+import os
+import argparse
```
Comment (blasten) [inline build/android_artifacts.py:14]: do these import need to be sorted?

**Review by blasten — APPROVED** [review]

File: build/android_artifacts.py (hunk 1 above)
Comment (godofredoc) [reply build/android_artifacts.py:9]: done!

File: build/android_artifacts.py (hunk 3 above)
Comment (godofredoc) [reply build/android_artifacts.py:14]: sorted.

File: build/android_artifacts.py (hunk 2 above)
Comment (godofredoc) [reply build/android_artifacts.py:15]: done!

**Pull request merged**